lib/db.js

```diff
@@ -600,37 +600,25 @@ module.exports = (
       return P.resolve()
     }
 
-    const tokenIds = sessionTokens.reduce((set, token) => {
-      if (token.createdAt <= Date.now() - TOKEN_PRUNING_MAX_AGE) {
-        set.add(token.id)
-      }
-      return set
-    }, new Set())
+    const tokenIds = sessionTokens
+      .filter(token => token.createdAt <= Date.now() - TOKEN_PRUNING_MAX_AGE)
+      .map(token => token.id)
 
-    const pruneCount = tokenIds.size
-    if (pruneCount === 0) {
+    if (tokenIds.length === 0) {
       return P.resolve()
     }
 
     return redis.update(uid, sessionTokens => {
-      sessionTokens = unpackTokensFromRedis(sessionTokens)
+      if (! sessionTokens) {
+        return
+      }
 
-      const keys = Object.keys(sessionTokens)
-      let keyCount = keys.length
-      let doneCount = 0
+      sessionTokens = unpackTokensFromRedis(sessionTokens)
 
-      keys.some(key => {
-        if (tokenIds.has(key)) {
-          delete sessionTokens[key]
-          keyCount -= 1
-          if (++doneCount === pruneCount) {
-            return true
-          }
-        }
-      })
+      tokenIds.forEach(id => delete sessionTokens[id])
```

Comment:
When I compared what I'd done in `pruneSessionTokens` to what I was doing in `deleteSessionToken`, I realised how inefficient and awkwardly long-winded my code in `pruneSessionTokens` was. Maybe it was late or something.

Anyway, simplified here.